rancherdesktop/version.go

```diff
@@ -0,0 +1,42 @@
+package rancherdesktop
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	"time"
+)
+
+// Version represents a version of the application. Present in both the config
+// passed to Upgrade Responder, and in the responses sent to clients.
+type Version struct {
+	// Must be a valid semver.
+	Name        string
+	ReleaseDate string
+	// Can be empty or valid semver.
+	MinUpgradableVersion string
```

Comment:
It is not clear to me what this field is used for. It looks like something that is passed from the config and included in the response, but what is the client supposed to do with it?

From the name of it, it sounds like the client needs to be at the `MinUpgradableVersion` or later to be able to upgrade to this version. But this should already be reflected in the `Supported` field, so why is this passed on in the response?

And if this is indeed an update constraint, should it be encoded in the `Rules` and not in the `Versions`?

Reply:
This is another holdover from upstream. Looking back, it isn't actually used there either. I'll remove it.